hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/volume/VolumeInfoStats.java

```diff
@@ -0,0 +1,146 @@
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.apache.hadoop.ozone.container.common.volume;
+
+import org.apache.hadoop.metrics2.MetricsSystem;
+import org.apache.hadoop.metrics2.annotation.Metric;
+import org.apache.hadoop.metrics2.annotation.Metrics;
+import org.apache.hadoop.metrics2.lib.DefaultMetricsSystem;
+import org.apache.hadoop.ozone.OzoneConsts;
+
+/**
+ * This class is used to track Volume Info stats for each HDDS Volume.
+ */
+@Metrics(about = "Ozone Volume Information Metrics",
+    context = OzoneConsts.OZONE)
+public class VolumeInfoStats {
+
+  private String metricsSourceName = VolumeInfoStats.class.getSimpleName();
+  private String volumeRootStr;
+  private HddsVolume volume;
+
+  private long spaceUsed;
+  private long spaceAvailable;
+  private long spaceReserved;
+
+
+  /**
+   * @param identifier Typically, path to volume root. e.g. /data/hdds
+   */
+  public VolumeInfoStats(String identifier, HddsVolume ref) {
+    this.metricsSourceName +=
+        '-' + identifier + StorageVolume.VolumeType.DATA_VOLUME.toString();
+    this.volumeRootStr = identifier;
+    this.volume = ref;
+    init();
+  }
+
+  public void init() {
+    MetricsSystem ms = DefaultMetricsSystem.instance();
+    ms.register(metricsSourceName, "Volume Info Statistics", this);
+  }
+
+  public void unregister() {
+    MetricsSystem ms = DefaultMetricsSystem.instance();
+    ms.unregisterSource(metricsSourceName);
+  }
+
+  @Metric("Metric to return the Storage Type")
+  public String getStorageType() {
+    return volume.getStorageType().toString();
+  }
+
+  @Metric("Returns the Directory name for the volume")
+  public String getStorageDirectory() {
+    return volume.getStorageDir().toString();
+  }
+
+  @Metric("Return the DataNode UID for the respective volume")
+  public String getDatanodeUuid() {
+    return volume.getDatanodeUuid();
+  }
+
+  @Metric("Return the Layout Version for the volume")
+  public int getLayoutVersion() {
+    return volume.getLayoutVersion();
+  }
+
+  @Metric("Returns the Volume Type")
+  public String getVolumeType() {
+    return volume.getType().name();
+  }
+
+  public String getMetricsSourceName() {
+    return metricsSourceName;
+  }
+
+  /**
+   * Test conservative avail space.
+   * |----used----|   (avail)   |++++++++reserved++++++++|
+   * |<------- capacity ------->|
+   * |<------------------- Total capacity -------------->|
+   * A) avail = capacity - used
+   * B) capacity = used + avail
+   * C) Total capacity = used + avail + reserved
+   */
+
+  /**
+   * Return the Storage type for the Volume.
+   */
+  @Metric("Returns the Used space")
+  public long getUsed() {
+    spaceUsed = volume.getVolumeInfo().getScmUsed();
+    return spaceUsed;
```

Comment:
```suggestion
    return volume.getVolumeInfo().getScmUsed();
```
The variable isn't needed. You can return the value directly.